railties/lib/rails/commands/server/server_command.rb

```diff
@@ -140,21 +95,72 @@ def restart_command
 
   module Command
     class ServerCommand < Base # :nodoc:
-      def help
-        puts Rails::Server::Options.new.option_parser(Hash.new)
+      DEFAULT_PID_PATH = File.expand_path("tmp/pids/server.pid").freeze
+
+      class_option :port, aliases: "-p", type: :numeric,
+        desc: "Runs Rails on the specified port.", banner: :port, default: 3000
+      class_option :binding, aliases: "-b", type: :string, default: "localhost",
+        desc: "Binds Rails to the specified IP.", banner: :IP
+      class_option :config, aliases: "-c", type: :string, default: "config.ru",
+        desc: "Uses a custom rackup configuration.", banner: :file
+      class_option :daemon, aliases: "-d", type: :boolean, default: false,
+        desc: "Runs server as a Daemon."
+      class_option :environment, aliases: "-e", type: :string,
+        desc: "Specifies the environment to run this server under (development/test/production).", banner: :name
+      class_option :pid, aliases: "-P", type: :string, default: DEFAULT_PID_PATH,
+        desc: "Specifies the PID file."
+      class_option "dev-caching", aliases: "-C", type: :boolean, default: nil,
+        desc: "Specifies whether to perform caching in development."
+
+      def initialize(*)
+        super
+        @server = args.shift
+        @log_stdout = options[:daemon].blank? && (options[:environment] || Rails.env) == "development"
       end
 
       def perform
         set_application_directory!
-
-        Rails::Server.new.tap do |server|
+        Rails::Server.new(server_options).tap do |server|
           # Require application after server sets environment to propagate
           # the --environment option.
           require APP_PATH
           Dir.chdir(Rails.application.root)
           server.start
         end
       end
+
+      def server_options
+        {
+          server:             @server,
+          log_stdout:         @log_stdout,
+          Port:               port,
+          Host:               host,
+          DoNotReverseLookup: true,
+          config:             options[:config],
+          environment:        environment,
+          daemonize:          options[:daemon],
+          pid:                options[:pid],
+          caching:            options["dev-caching"],
+          restart_cmd:        restart_command
+        }
+      end
+
+      private
+        def port
+          ENV.fetch("PORT", options[:port]).to_i
+        end
+
+        def host
+          ENV.fetch("HOST", options[:binding])
+        end
+
+        def environment
+          options[:environment] || Rails::Command.environment
+        end
+
+        def restart_command
+          "bin/rails server #{ARGV.join(' ')}"
```

Comment:
In this context, the `args` variable does not contain any arguments. 
Therefore, kept the argument in the constructor and used it.